### ISSUE TITLE
hasManyToMany count relation not working properly

### ISSUE THREAD

**Issue by bkwint** [phalcon/cphalcon#1945]
When using the count[Relation] magic method for a model, many to many relations are not counted, instead when count[Relation] is called a Phalcon\Mvc\Model\Resultset\Simple is returned. For a hasMany relation this does work.

Example classes:
~~~
namespace Models\Translation;

use Phalcon\Mvc\Model;

class Language extends Model
{
    /**
     * The tablename of the table we are interfacing with
     */
    const TABLE_NAME = 'Translation_Language';
    
    /**
     * @var string $_iso Iso language code
     */
    protected $iso;
    /**
     * @var string $_fullname Full language name
     */
    protected $fullname;

    /**
     * The table name for this model should be returned
     * by this function
     *
     * @return string Table name
     */
    public function getSource()
    {
        return self::TABLE_NAME;
    }
    /**
     * On initialize we prepare the model
     * 
     *  - Build relations
     */
    public function initialize()
    {
        $this->hasMany('iso', '\Models\Translation\Message', 'language', array(
            'alias' => 'translations'
        ));
        
        $this->hasManyToMany(
            'iso', 
            '\Models\Translation\Message', 
            'language', 
            'id', 
            '\Models\Translation\SourceMessage', 
            'id', 
            array(
                'alias' => 'sourceMessages'
            )
        );
    }
}

class Message extends Model
{
    /**
     * The name of the database table we are interfacing with
     */
    const TABLE_NAME = 'Translation_Message';
    
    /**
     * The id of the source message we are interfacing with
     * 
     * @var int
     */
    protected $id;
    /**
     * The iso code for the language we are working with
     * 
     * @var string
     */
    protected $language;
    /**
     * The actual translation for the value
     * 
     * @var string 
     */
    protected $translation;
    
    /**
     * What is the name of the table we should be interfacing with
     * 
     * @return string
     */
    public function getSource()
    {
        return self::TABLE_NAME;
    }
    /**
     * On initialize we define the relations that this model is supposed
     * to have.
     */
    public function initialize()
    {
        $this->hasOne('language', '\Models\Translation\Language', 'iso', array(
            'alias' => 'lang'
        ));
    }
}

class SourceMessage extends Model
{
    /**
     * The database table we are interfacing with
     */
    const TABLE_NAME = 'Translation_SourceMessage';
    
    /**
     * Message identifier
     * 
     * @var int
     */
    protected $id;
    /**
     * Category for grouping messages to specific parts of the system
     * 
     * @var string
     */
    protected $category;
    /**
     * The message that needs to be translated
     * 
     * @var string
     */
    protected $message;
    
    /**
     * Sets the tablename for the model
     * 
     * @return string
     */
    public function getSource()
    {
        return self::TABLE_NAME;
    }
    /**
     * Sets up the model for use
     */
    public function initialize()
    {
        $this->hasMany('id', '\Models\Translation\Message', 'id', array(
            'alias' => 'translations'
        ));

        $this->hasManyToMany(
            'id', 
            '\Models\Translation\Message', 
            'id', 
            'language', 
            '\Models\Translation\Language', 
            'iso', 
            array(
                'alias' => 'languages'
        ));
    }
}
~~~

Database defintion:
~~~
CREATE TABLE IF NOT EXISTS `Translation_SourceMessage` (
  `id` INT(11) NOT NULL AUTO_INCREMENT,
  `category` VARCHAR(32) NOT NULL,
  `message` TEXT NOT NULL,
  PRIMARY KEY (`id`))
ENGINE = InnoDB
AUTO_INCREMENT = 3
DEFAULT CHARACTER SET = latin1;

CREATE TABLE IF NOT EXISTS `Translation_Language` (
  `iso` VARCHAR(7) NOT NULL,
  `fullname` VARCHAR(45) NOT NULL,
  PRIMARY KEY (`iso`))
ENGINE = InnoDB;

CREATE TABLE IF NOT EXISTS `Translation_Message` (
  `id` INT(11) NOT NULL DEFAULT '0',
  `language` VARCHAR(7) NOT NULL,
  `translation` TEXT NOT NULL,
  PRIMARY KEY (`id`, `language`),
  INDEX `fk_Translation_Message_Translation_Language1_idx` (`language` ASC),
  CONSTRAINT `FK_Message_SourceMessage`
    FOREIGN KEY (`id`)
    REFERENCES `Translation_SourceMessage` (`id`)
    ON DELETE CASCADE,
  CONSTRAINT `fk_Translation_Message_Translation_Language1`
    FOREIGN KEY (`language`)
    REFERENCES `Translation_Language` (`iso`)
    ON DELETE CASCADE
    ON UPDATE CASCADE)
ENGINE = InnoDB
DEFAULT CHARACTER SET = latin1;
~~~

Example code:
~~~
$language = Language::findFirst();
$sourceMessage = $language->countSourceMessages(); //Phalcon\Mvc\Model\Resultset\Simple is returned in stead of an integer
~~~

**Comment by Rigiytip** [comment]
+1

**Comment by mapirelli** [comment]
i got the same error in version 1.3

$this->hasManyToMany('id', 'RegistryAnswer', 'fk_registry', 'fk_answer', 'Answer', 'id', array('alias' => 'Answers'));

public function __construct(Registry $registry,$relations = ['Answers'])
{
        $this->id = $registry->id;
        ...
        $this->answersCorrect = $registry->countAnswers();
}

{{ answersCorrect }} -> fatal error

[01-Jul-2015 09:22:40 Europe/Berlin] PHP Catchable fatal error:  Object of class Phalcon\Mvc\Model\Resultset\Simple could not be converted to string in /Users/mariano.pirelli/PHP/Concorsi/apps/cache/_users_mariano.pirelli_php_concorsi_apps_backend_views_index_registries.phtml.php on line 38

**Comment by sergeyklay** [comment]
Fixed in 2.1.x branch
